src/services/CapabilitiesManager.ts

```diff
@@ -10,27 +10,59 @@ import { t } from '@nextcloud/l10n'
 import { getRemoteCapabilities } from './federationService.ts'
 import BrowserStorage from '../services/BrowserStorage.js'
 import { useTalkHashStore } from '../stores/talkHash.js'
-import type { Capabilities, JoinRoomFullResponse } from '../types'
+import type { Capabilities, Conversation, JoinRoomFullResponse } from '../types'
 
 type Config = Capabilities['spreed']['config']
-type RemoteCapabilities = Record<string, Capabilities & Partial<{ hash: string }>>
+type RemoteCapability = Capabilities & Partial<{ hash: string }>
+type RemoteCapabilities = Record<string, RemoteCapability>
+type TokenMap = Record<string, string|null>
+
+let remoteTokenMap: TokenMap = generateTokenMap()
 
 const localCapabilities: Capabilities = _getCapabilities() as Capabilities
 const remoteCapabilities: RemoteCapabilities = restoreRemoteCapabilities()
 
+/**
+ * Generate new token map based on remoteCapabilities and cachedConversation
+ */
+function generateTokenMap() {
+	const tokenMap: TokenMap = {}
+	const storageValue = BrowserStorage.getItem('cachedConversations')
+	if (!storageValue?.length) {
+		return {}
+	}
+	const cachedConversations = JSON.parse(storageValue) as Conversation[]
+	cachedConversations.forEach(conversation => {
+		tokenMap[conversation.token] = conversation.remoteServer || null
+	})
+
+	return tokenMap
+}
```

Comment:
Why not keep storing tokens in 'remoteCapabilities' storage. It will only parse fed conversations tokens.

Reply:
to distinguigh local (`null`), remote (`string`) and unknown (`undefined`) conversations